### ISSUE TITLE
Convert the rego policies to be violated by default

### ISSUE THREAD

**Issue by noamd-legit** [Legit-Labs/legitify#84]
### TL;DR

Currently, policies are not violated by default, and we mark them as violated only if certain conditions are met. This makes our policies more complex and prone to errors.
For example:
```
default missing_default_branch_protection_force_push = false

missing_default_branch_protection_force_push {
    missing_default_branch_protection
}

missing_default_branch_protection_force_push {
        ....
	count(rules_allow_force_push) > 0
}
```
The `missing_default_branch_protection_force_push` policy needs to be defined as a partial policy that first checks if branch protection exists and then the specific condition. 

### Detailed design

Instead of the above, define the policy like this:

```
default missing_default_branch_protection_force_push = true
missing_default_branch_protection_force_push = false {
        ....
	count(rules_allow_force_push) == 0
}
```

This way, the policy needs to check only for the good case and not for all of the bad options.

### Additional information

* Make the changes gradually -> Pull request per rego file
* The tests shouldn't fail. If you find yourself changing the tests, make sure you didn't break anything

**Comment by morelmiles** [comment]
Hi @noamd-legit, 
Do you mind unblocking me as I solve this? When I make the changes that you suggested above, a number of tests (Gitlab) fail . See screenshot:
![image](https://user-images.githubusercontent.com/48476293/218140831-d1783394-bb70-41ff-a1b5-0d66e77e9b4c.png)

This is the error that I get all through. See screenshot

![image](https://user-images.githubusercontent.com/48476293/218142133-876134bf-3d49-46bf-9334-fb16e207d5a2.png)
